Neo4j configure auth

mneedham opened the pull request:
This PR makes auth configurable and sets the default to enabled rather than disabled. 

Also updated the notes to explain how to set a new password.

Commit:
Make auth configurable

unguiculus commented on incubator/neo4j/templates/NOTES.txt:
```diff
@@ -1,9 +1,35 @@
+
+We can see the content of the logs by running the following command:
+
+kubectl logs -l "app=neo4j,component=core"
```
Don't hard-code the app label:
-> `kubectl logs -l "app={{ template "neo4j.name" . }},component=core"`

unguiculus commented on incubator/neo4j/templates/tests/test-config.yaml:
```diff
@@ -15,10 +15,12 @@ data:
         while true; do
-          echo "testing http://$name:7474/db/data"
-          out=`wget http://$name:7474/db/data -O - 2>&1`
+          kubectl exec $name -- bin/cypher-shell -u neo4j -p neo4j "CALL dbms.security.changePassword('newPassword')"
```
Tests should not change the password.

unguiculus reviewed (changes requested):
It should be possible to specify or auto-generate the password instead of having to change it after the installation. Tests should then use this password instead of changing it. Use a secret for the password. Here's an example: https://github.com/kubernetes/charts/blob/master/stable/mariadb/templates/secrets.yaml

mneedham commented:
@unguiculus just got around to addressing your suggestions. I've now implemented something similar to the Maria DB one with auto generated password stored in secrets. 

Let me know if there's anything else I should change.

mneedham commented:
@unguiculus Pinging to see if there's anything else I need to do :)

unguiculus commented on incubator/neo4j/templates/secrets.yml:
```diff
@@ -0,0 +1,18 @@
+metadata:
+  name: {{ template "neo4j.secrets.fullname" . }}
+  labels:
+    app: {{ template "neo4j.secrets.fullname" . }}
```
`app: {{ template "neo4j.name" . }}`

unguiculus commented on incubator/neo4j/templates/NOTES.txt:
```diff
@@ -1,9 +1,25 @@
+We can now run a query to find the topology of the cluster.
+
+kubectl run -it --rm cypher-shell \
+    --image=neo4j:3.2.1-enterprise \
```
`imageTag` is `3.2.3`

unguiculus reviewed (changes requested):
Sorry for the delay. We're almost there.

mneedham commented:
@unguiculus done!

unguiculus commented on incubator/neo4j/templates/secrets.yml:
```diff
@@ -0,0 +1,18 @@
+metadata:
+  name: app: {{ template "neo4j.name" . }}
+  labels:
+    app: app: {{ template "neo4j.name" . }}
```
One `app:` too many

mneedham replied:
@unguiculus ooops, sorry. Have tried again.

unguiculus commented on incubator/neo4j/templates/secrets.yml:
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: app: {{ template "neo4j.name" . }}
```
You should have only changed the `app` label, not the name. Please change it back to `{{ template "neo4j.secrets.fullname" . }}`

unguiculus reviewed (changes requested):
Please don't squash commits when updating the PR. Github can't show the changes then. The PR will be squashed anyway when it is merged.

Commit:
fixing typos

unguiculus committed:
Rename secrets.yml to secret.yml

unguiculus commented:
/ok-to-test

unguiculus committed:
Bump chart version

unguiculus commented:
I installed the chart successfully and ran tests on my GKE cluster. Tests seems to never finish. Here's the test log I get:

```
$ stern misty-bat-neo4j-service-test-1xcb3                                                                                      ± ★ ✎ ⫘ (97e8557)
+ misty-bat-neo4j-service-test-1xcb3 › misty-bat-ui-test
misty-bat-neo4j-service-test-1xcb3 misty-bat-ui-test 1..2
misty-bat-neo4j-service-test-1xcb3 misty-bat-ui-test ok 1 Testing Neo4j cluster has quorum
misty-bat-neo4j-service-test-1xcb3 misty-bat-ui-test ok 2 Testing we can get the cluster role of each server
```

unguiculus commented:
@mneedham Did you get a chance to look at the tests?

mneedham commented:
@unguiculus I've just tried it locally using these commands:

```
$ helm init && kubectl rollout status -w deployment/tiller-deploy --namespace=kube-system
$ helm install incubator/neo4j --wait --name neo-helm

$ helm test neo-helm
RUNNING: neo-helm-neo4j-service-test-gmp9b
PASSED: neo-helm-neo4j-service-test-gmp9b
```

And it seems happy. I have similar logs to the ones you're seeing:

```
$ kubectl logs neo-helm-neo4j-service-test-gmp9b
1..2
ok 1 Testing Neo4j cluster has quorum
ok 2 Testing we can get the cluster role of each server
```

Commit:
renaming to make lint happy

mneedham commented:
The lint checker is unhappy with the yml prefix so I've renamed the secret file

unguiculus commented:
Please update to latest master. This will probably fix the linting issue.

unguiculus commented:
/retest

mneedham commented:
I've rebased against master

unguiculus committed:
Empty commit to trigger CircleCI

unguiculus committed:
Fix linting issue in values.yaml

unguiculus commented:
I ran Helm tests successfully on my cluster. CI probably just times out.

unguiculus commented:
/lgtm

unguiculus commented:
I think you could move the chart to stable.
